src/sentry/plugins/sentry_mail/models.py

```diff
@@ -275,7 +278,16 @@ def notify(self, notification):
         }
 
         for user_id in self.get_send_to(project=project, event=event):
-            self.add_unsubscribe_link(context, user_id, project)
+            unsubscribe_link = generate_signed_link(
+                user_id,
+                'sentry-account-email-unsubscribe-project',
+                kwargs={
+                    'project_id': project.id,
+                }
+            )
+
+            context['unsubscribe_link'] = unsubscribe_link + '&source=alert_email'
```

Comment:
feels brittle to modify links once they've been generated by a helper function. would rather pass `source` as additional params to the function which does the work.